### PR TITLE
Added some code-style guidelines to CONTRIBUTING

### PR DESCRIPTION
**Pull request by TheBusyBiscuit**
## Description

I am proposing some changes to the `CONTRIBUTING.md`.
Currently my advice has always been "stick to the current style".
But due to the ongoing increase in contributions, it may be wise to facilitate some concrete dos and don'ts.

Anything you wanna change or add?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -94,3 +94,48 @@ Then you should be able build it via Maven using the goals `clean package`.
...
+## :black_nib: Code Style guidelines
+The general gist when it comes to code style: **Try to be consistent!**.<br>
+Try to stay inline with the code that surrounds you, having an entire package or even a single file that's filled with plenty of different and inconsistent code styles is just hard to read or maintain. That's why we wanna make sure everyone follows these simple principles:
+
+1. Imports
+* Don't use wildcard (`*`) imports!
+* Don't import unused classes!
+* Don't use static imports!
+2. Annotations
+* Methods and parameters should be annotated with `@Nullable` or `@Nonnull`!
```
Comment (LinoxGH) [inline CONTRIBUTING.md:107]: Should also tell from which api/package.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -94,3 +94,48 @@ Then you should be able build it via Maven using the goals `clean package`.
...
+* Don't use static imports!
+#### 2. Annotations
+* Methods and parameters should be annotated with `@Nullable` or `@Nonnull`!
+* Methods that override a method must be annotated with `@Override`!
+* Interfaces with only one method should be annotated using `@FunctionalInterface`!
+* If you deprecate a method, add an `@deprecated` section to the javadocs explaining why you did it.
+#### 3. Documentation
+* Every class and every public method should have a Javadocs section assigned to it.
+* New packages should have a `package-info.java` file with documentation about the package.
+* Classes should have an `@author` tag.
```
Comment (LinoxGH) [inline CONTRIBUTING.md:114]:
```suggestion
* Classes should have an `@author` tag.
* Classes should have `@see` tags pinpointing the apis used in the class.
```

**Review by LinoxGH — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (TheBusyBiscuit) [reply CONTRIBUTING.md:107]: Yes!

File: CONTRIBUTING.md (hunk 2 above)
Comment (TheBusyBiscuit) [reply CONTRIBUTING.md:114]: Hmm, I will include this suggestion but I'll probably rephrase it a little.

**Review by poma123 — CHANGES_REQUESTED** [review]
Maybe something about putting `else {`, `else if {`, `catch {` etc. on a new line?

**Comment by TheBusyBiscuit** [comment]
> Maybe something about putting `else {`, `else if {`, `catch {` etc. on a new line?

Thought about that but I decided against it first, many people seem to put them on the same line.
So I wouldn't have a problem with changing it to that style of brackets, we would of course need to update all the code before though.
So yeah, I'm currently leaving that open, what's your opinion on it?

**Comment by LinoxGH** [comment]
> > Maybe something about putting `else {`, `else if {`, `catch {` etc. on a new line?
> 
> Thought about that but I decided against it first, many people seem to put them on the same line.
> So I wouldn't have a problem with changing it to that style of brackets, we would of course need to update all the code before though.
> So yeah, I'm currently leaving that open, what's your opinion on it?

I personally prefer it on the same line.

**Comment by poma123** [comment]
> Thought about that but I decided against it first, many people seem to put them on the same line.
> So I wouldn't have a problem with changing it to that style of brackets, we would of course need to update all the code before though.
> So yeah, I'm currently leaving that open, what's your opinion on it?

Well, I've always put them on the same line so... Although putting them on a new line seems clean as well, I don't think that putting them on the same line would be better or worse.

**Comment by TheBusyBiscuit** [comment]
I don't really mind which of the two we follow, as long as it is consistent. Looks like we got two people in favour of same-line, @Sfiguz7 what about you?

**Comment by TheBusyBiscuit** [comment]
I'll go with the OTBS standard then, so else/else if now go on the same line, since that's what some people seem to prefer.
I'll update master in a second too

**Review by Sfiguz7 — APPROVED** [review]
It seems I'm always late to the party uff
I am used to OTBS as well but I didn't mind using new lines as plenty of times it was clearer. Someone is going to have fun cleaning every single damn class (or we could just PR a bunch at a time by maybe notifying others of who is taking what just so we can split the burden)

**Review by poma123 — APPROVED** [review]

**Comment by TheBusyBiscuit** [comment]
> It seems I'm always late to the party uff
> I am used to OTBS as well but I didn't mind using new lines as plenty of times it was clearer. Someone is going to have fun cleaning every single damn class (or we could just PR a bunch at a time by maybe notifying others of who is taking what just so we can split the burden)

I already commited that >.>

**Review by LinoxGH — DISMISSED** [review]

**Comment by Seggan** [comment]
Don't overuse inverted if + `return` statements in listeners.

**Comment by LinoxGH** [comment]
Maybe something about putting spaces in array initializers, and in if statements between brackets and etc.

**Comment by TheBusyBiscuit** [comment]
@LinoxGH That is already in there under "horizontal whitespaces"

**Review by poma123 — APPROVED** [review]

**Review by Sfiguz7 — APPROVED** [review]

**Review by LinoxGH — APPROVED** [review]

**Pull request merged**